kb(Grid): add kb for allow/prevent a row selection (#2087)
telerik/blazor-docs: knowledge-base/grid-allow-prevent-row-selection.md

type: how-to
page_title: Allow/Prevent Selection of Certain Row in the Grid
slug: grid-kb-prevent-row-selection
tags: grid, selection
ticketid: 1441291
res_type: kb
---

## Environment

<table>
    <tbody>
        <tr>
            <td>Product</td>
            <td>Grid for Blazor</td>
        </tr>
    </tbody>
</table>

## Description

This KB article answers the following questions:

* How to disable the selection of a row in the Grid?
* How to prevent the selection of a certian row in the Grid?
* How to disable the checkbox column of a disabled row in a Telerik Blazor Grid?
* How to customize the checkbox column of a certain row?
* How to prevent the selection of Grid items conditionally through the [SelectedItemsChanged]({%slug grid-events%}#selecteditemschanged) event?

## Solution

The key points in the required implementation are:

* Use the [OnRowRender]({%slug grid-events%}#onrowrender) handler to set the custom class to the desired row. 
* Use the `SelectedItemsChanged` event of the Grid to prevent the selection of items conditionally.

> * Disable the checkboxes by conditionally applying classes to the rows in which the checkboxes should be disabled.

Reviewer: Avoid [using ambiguous modal verbs](https://docs.telerik.com/style-guide/basic-rules-and-guidelines#use-properly-should-could-must-have-to-need-to) like "should".
```suggestion
* Disable the checkboxes by conditionally applying classes to the rows where the checkboxes will be disabled.
```

> * Add a CSS rule that targets those checkboxes within the rows with the applied custom class and disabled them.

Reviewer: ```suggestion
* Add a CSS rule that targets and disables the checkboxes on the rows where you applied custom classes.
```

             SelectedItems="@SelectedItems"
             SelectedItemsChanged="@((IEnumerable<SampleData> items) => SelectedItemsChanged(items))"
             SelectionMode="@GridSelectionMode.Multiple"
             OnRowRender="@OnRowRenderHandler"
             EditMode="GridEditMode.Incell">
    <GridColumns>
        <GridCheckboxColumn SelectAll="@true" />
        <GridColumn Field="@(nameof(SampleData.Id))" />
        <GridColumn Field="@(nameof(SampleData.Name))" Title="Employee Name" />
        <GridColumn Field="@(nameof(SampleData.Team))" Title="Team" />
    </GridColumns>
</TelerikGrid>

<style>
@* CSS that targets the checkbox column of the desired rows *@
    .unselectable-row .k-checkbox {
        opacity: 0.5;
        pointer-events: none;
    }
</style>

@code {
    
    private void OnRowRenderHandler(GridRowRenderEventArgs args)
    {
        var item = args.Item as SampleData;
        // applying a custom class conditionally 
        if (item.Name.Contains("3") || item.Name.Contains("4"))
        {
            args.Class = "unselectable-row";
        }
    }

    private List<SampleData> SelectedItems { get; set; } = new List<SampleData>();

>     // select only the selectable item in the SelectedItemsChanged handler

Reviewer: Use the appropriate capitalization and punctuation when writing complete sentences in the comments.
```suggestion
    // Select only the selectable item in the SelectedItemsChanged handler.
```
